pkg/pgmodel/migrations/sql/idempotent/base.sql

```diff
@@ -1123,26 +1150,100 @@ BEGIN
             SELECT label_id
             FROM unnest($1) as labels(label_id)
             WHERE NOT EXISTS (
-                 SELECT 1
-                 FROM  SCHEMA_DATA_SERIES.%1$I series_exists
-                 WHERE series_exists.labels && ARRAY[labels.label_id]
-                 LIMIT 1
+                SELECT 1
+                FROM  SCHEMA_CATALOG.series series_exists
+                WHERE series_exists.labels && ARRAY[labels.label_id]
+                LIMIT 1
             )
         )
         DELETE FROM SCHEMA_CATALOG.label
         WHERE id IN (SELECT * FROM confirmed_drop_labels) AND key != '__name__';
     $query$, metric_table) USING label_array;
 
-   IF SCHEMA_CATALOG.is_timescaledb_installed() THEN
-        PERFORM drop_chunks(table_name=>metric_table, schema_name=> 'SCHEMA_DATA', older_than=>older_than, cascade_to_materializations=>FALSE);
-   ELSE
-        EXECUTE format($$ DELETE FROM SCHEMA_DATA.%I WHERE time < %L $$, metric_table, older_than);
-   END IF;
-   RETURN true;
+    -- wait for current insertions to be done, and ensure that all future
```

Comment:
is it worth avoiding this lock and the epoch increment if there are no series rows where delete_epoch IS NOT NULL?

Reply:
@cevian I think the logic would have to be
```sql
count(deleted_series) = 0 AND count(delete_epoch IS NOT NULL) = 0
```
but I think we should hold off for now: this is already getting complicated to test, and I don't think the once-an-hour barrier will be that large a performance issue